.github/workflows/codeql.yml

```diff
@@ -22,56 +22,91 @@ on:
-    analyze:
-        name: Analyze (${{ matrix.language }})
-        runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    # C# CodeQL Analysis
+    analyze-csharp:
+        name: Analyze (csharp)
+        runs-on: ubuntu-latest
         permissions:
             security-events: write
 
-        strategy:
-            fail-fast: false
-            matrix:
-                include:
-                    - language: csharp
-                      build-mode: manual
-                    - language: go
-                      build-mode: manual
-                    - language: java-kotlin
-                      build-mode: manual
-                    - language: javascript-typescript
-                      build-mode: none
-                    - language: python
-                      build-mode: none
-                    - language: rust
-                      build-mode: none
-                    - language: c-cpp
-                      build-mode: manual
-                    - language: actions
-                      build-mode: none
-
         steps:
             - name: Checkout repository
               uses: actions/checkout@v4
 
+            - name: Check for C# file changes
+              if: github.event_name == 'pull_request'
+              uses: dorny/paths-filter@v3
+              id: changes
+              with:
+                  filters: |
+                      csharp:
+                          - 'csharp/**'
+                          - 'glide-core/src/**'
+                          - 'glide-core/redis-rs/redis/src/**'
+                          - '.github/workflows/codeql.yml'
+
             - name: Initialize CodeQL
+              if: github.event_name != 'pull_request' || steps.changes.outputs.csharp == 'true'
               uses: github/codeql-action/init@v3
               with:
-                  languages: ${{ matrix.language }}
-                  build-mode: ${{ matrix.build-mode }}
+                  languages: csharp
+                  build-mode: manual
                   config-file: .github/codeql/codeql-config.yml
 
-            # C++ Build
-            - name: Build C++ components
-              if: matrix.language == 'c-cpp'
+            - name: Setup .NET
+              if: github.event_name != 'pull_request' || steps.changes.outputs.csharp == 'true'
+              uses: actions/setup-dotnet@v4
+              with:
+                  dotnet-version: "8.0.x"
+
+            - name: Build C# components
+              if: github.event_name != 'pull_request' || steps.changes.outputs.csharp == 'true'
               shell: bash
               run: |
-                  cd glide-core
-                  cargo build --release
+                  cd csharp
+                  dotnet build sources/Valkey.Glide/Valkey.Glide.csproj --configuration Lint --framework net8.0
+                  dotnet build tests/Valkey.Glide.UnitTests/Valkey.Glide.UnitTests.csproj --configuration Lint --framework net8.0
```

Comment:
Do we not also need to run CodeQL for Integration Tests which is a separate project from UnitTests?